flask_static_digest/__init__.py

```diff
@@ -61,26 +59,38 @@ def static_url_for(self, endpoint, **values):
         :param values: Arguments of the URL rule
         :return: Static file path.
         """
-        if not self.has_manifest:
-            if self.host_url:
-                return self._prepend_host_url(self.host_url,
-                                              values.get("filename"))
-            else:
-                return flask_url_for(endpoint, **values)
-
         new_filename = {}
         filename = values.get("filename")
 
+        # note: this is taken from flask's url_for
+        # ( resolves relative endpoints )
+        if request is not None:
+            blueprint_name = request.blueprint
+
+            # If the endpoint starts with "." and the request matches a
+            # blueprint, the endpoint is relative to the blueprint.
+            if endpoint[:1] == ".":
+                if blueprint_name is not None:
+                    endpoint = f"{blueprint_name}{endpoint}"
+                else:
+                    endpoint = endpoint[1:]
+        # endnote
+
         if filename:
             # If the manifest lookup fails then use the original filename
             # so that Flask doesn't throw a 500, but instead a proper 404.
             # The above only happens if your template has an invalid filename.
-            new_filename["filename"] = self.manifest.get(filename, filename)
+            manifest = self.manifests.get(endpoint)
+            if manifest:
+                new_filename["filename"] = manifest.get(filename, filename)
 
         merged_values = {**values, **new_filename}
 
-        if self.host_url:
-            return self._prepend_host_url(self.host_url,
-                                          merged_values.get("filename"))
-        else:
-            return flask_url_for(endpoint, **merged_values)
+        flask_url = flask_url_for(endpoint, **merged_values)
+        return urljoin(self.host_url, flask_url)
+
+    def _load_manifest(self, scaffold, manifest_path):
```

Comment:
Given the other proposed changes can you change the `scaffold` argument to `blueprint` here?

Reply:
[0501677 renaming scaffold and ingestScaffold, reworked load_manifest](https://github.com/nickjj/flask-static-digest/pull/33/commits/05016779a5593dd065231c662fd6040c2fb6e09e)